src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,397 @@
+        self.commandDelegate.send(result, callbackId: command.callbackId)
+    }
+
+    @objc(logIn:) func logIn(command: CDVInvokedUrlCommand) {
+        guard let appUserID = command.arguments[0] as? String else {
+            self.sendBadParameterFor(command: command, parameterNamed: "appUserID", expectedType: NSString.self)
+            return
+        }
+
+        RCCommonFunctionality.logIn(withAppUserID: appUserID,
+                                    completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(logOut:) func logOut(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.logOut(completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+
+    @objc(setupShouldPurchasePromoProductCallback:) func setupShouldPurchasePromoProductCallback(command: CDVInvokedUrlCommand) {
+        self.shouldPurchasePromoProductCallbackID = command.callbackId;
+        // TODO: This next line is needed, right? It wasn't in the old version.
+        self.sendOKFor(command: command)
+    }
+
+    @objc(setDebugLogsEnabled:) func setDebugLogsEnabled(command: CDVInvokedUrlCommand) {
+        guard let debugLogsEnabled = (command.arguments[0] as? NSNumber)?.boolValue else {
+            self.sendBadParameterFor(command: command, parameterNamed: "debugLogsEnabled", expectedType: NSNumber.self)
+            return
+        }
+
+        RCCommonFunctionality.setDebugLogsEnabled(debugLogsEnabled)
+        self.sendOKFor(command: command)
+    }
+
+    @objc(getPurchaserInfo:) func getPurchaserInfo(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.getPurchaserInfo(completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(syncPurchases:) func syncPurchases(command: CDVInvokedUrlCommand) {
+        RCCommonFunctionality.syncPurchases(completionBlock: self.responseCompletion(forCommand: command))
+    }
+
+    @objc(setAutomaticAppleSearchAdsAttributionCollection:) func setAutomaticAppleSearchAdsAttributionCollection(command: CDVInvokedUrlCommand) {
+        guard let automaticCollection = (command.arguments[0] as? NSNumber)?.boolValue else {
+            self.sendBadParameterFor(command: command,
+                                     parameterNamed: "AutomaticAppleSearchAdsAttributionCollection",
+                                     expectedType: NSNumber.self)
+            return
+        }
+
+        RCCommonFunctionality.setAutomaticAppleSearchAdsAttributionCollection(automaticCollection)
+        self.sendOKFor(command: command)
+    }
+
+    @objc(setSimulatesAskToBuyInSandbox:) func setSimulatesAskToBuyInSandbox(command: CDVInvokedUrlCommand) {
+        guard let askToBuyInSandbox = (command.arguments[0] as? NSNumber)?.boolValue else {
+            self.sendBadParameterFor(command: command,
+                                     parameterNamed: "setSimulatesAskToBuyInSandbox",
+                                     expectedType: NSNumber.self)
```

Comment:
I genuinely wonder how many times we might have messed up the parameters for these before. But I'm scared to find out

Reply:
Same. Ha.